apps/total-typescript/src/templates/book-chapter-template.tsx

```diff
@@ -154,21 +172,24 @@ const BookChapterTemplate = ({
                     )
                   },
                   h2: (props: any) => {
+                    const resourceId = resources?.find(
+                      (r) => r.slug === props.id,
+                    )?._id
                     return (
                       <BookmarkableMarkdownHeading
                         onAddBookmark={handleAddBookmark}
                         appendValueForRepeatedIds={`-for-${slugify(
                           chapter.title,
                         )}`}
                         as="h2"
+                        resourceId={resourceId}
                         {...props}
                       />
                     )
                   },
                   h3: (props: any) => {
                     return (
                       <BookmarkableMarkdownHeading
-                        onAddBookmark={handleAddBookmark}
                         appendValueForRepeatedIds={`-for-${slugify(
```

Comment:
theres no way currently (that I can think of) h3s and h4s to set the parent h2 `resourceId` for h3s and h4s so we don't pass the `onAddBookmark` prop so it won't render the bookmark button